src/theme/components/toggle.ts

```diff
@@ -0,0 +1,66 @@
+import { helperTextMargin } from "./global";
+
+// Style object for the Checkbox's label
+const baseStyleLabel = {
+  fontSize: "sm",
+  marginBottom: "0",
+  marginLeft: "xs",
+};
+
+// Style object for the Checkbox's helper text
+const baseStyleHelper = {
+  ...helperTextMargin,
+  marginLeft: "xs",
+};
+
+const baseStyle = {
+  label: baseStyleLabel,
+  helper: baseStyleHelper,
+};
+
+const Switch = {
+  baseStyle: {
+    opacity: 0.4,
+    track: {
+      p: "4px",
+    },
+  },
+  variants: {
+    standard: {
+      track: {
+        _checked: {
+          bg: "ui.link.primary",
+          opacity: 1,
+        },
+      },
+    },
+    disabled: {
+      track: {
+        bg: "ui.gray.medium",
+      },
+    },
+    error: {
```

Comment:
This doesn't need to be a variant. When you pass in the `isInvalid` prop to a Chakra component, you can update that state by targeting the `_invalid` key. Above in the `baseStyle`, you can add:
```
baseStyle: {
  opacity: 0.4,
  track: {
    p: "4px",
  },
  _invalid: {
    track: {
      bg: "red.200",
      _checked: {
        bg: "red.500",
      },
    },
  },
},
```

Reply:
You might be able to write everything without needing to have variants involved.